### PR TITLE
Sumo monitors installation as a part of the setup job

### PR DESCRIPTION
**Pull request by starzu-sumo**
##### Description

In order to simplify k8s onboarding in Sumo Logic, we want to install monitors automatically from the setup job. 

Behaviour:
* the monitors will be installed in the `Kuberenetes` monitors folder if it doesn't exist yet
* the monitors installation is enabled by default and can be disabled with `sumologic.setup.monitors.enabled`
* the monitors are enabled by default and can be disabled with `sumologic.setup.monitors.monitorStatus`
* there are no notifications for the monitors by default
* email notifications can be enabled with `sumologic.setup.monitors.notificationEmails`

##### Checklist

- [x] Changelog updated

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: ci/shellcheck.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 set -e
 
 echo "Checking the bash scripts with shellcheck..."
-find . ! -path '*deploy/helm/sumologic/conf/setup/setup.sh' ! -path "*/tmp/*" -name '*.sh' -type 'f' -print |
+find . ! -path '*deploy/helm/sumologic/conf/setup/setup.sh' ! -path '*deploy/helm/sumologic/conf/setup/monitors.sh' ! -path "*/tmp/*" -name '*.sh' -type 'f' -print |
```
Comment (sumo-drosiek) [inline ci/shellcheck.sh:6]: Could you add analogical check like for `setup.sh` (is below)?

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -122,6 +122,16 @@ sumologic:
     ## uncomment for the debug mode (disables the automatic run of the setup.sh script)
     # debug: true
 
+    monitors:
+      ## If enabled, a pre-install hook will create k8s monitors in Sumo Logic
+      setupEnabled: true
+
+      ## If true, the installed monitors will be disabled by default
+      monitoringDisabled: false
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:130]: I would change this flag to `default` `enableByDefault` or sth like that. There is no need to repeat `monitoring` keyword here

File: deploy/helm/sumologic/conf/setup/monitors.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      -var="email_notifications_critical=[{${NOTIFICATIONS_SETTINGS},${NOTIFICATIONS_CONTENT},run_for_trigger_types=[\"Critical\", \"ResolvedCritical\"]}]" \
+      -var="email_notifications_warning=[{${NOTIFICATIONS_SETTINGS},${NOTIFICATIONS_CONTENT},run_for_trigger_types=[\"Warning\", \"ResolvedWarning\"]}]" \
+      -var="email_notifications_missingdata=[{${NOTIFICATIONS_SETTINGS},${NOTIFICATIONS_CONTENT},run_for_trigger_types=[\"MissingData\", \"ResolvedMissingData\"]}]" \
+{{- end }}
+      || { echo "Error during applying Terraform monitors."; exit 1; }
+else
+  echo "The monitors were already installed in ${MONITORS_FOLDER_NAME}."
+  echo "You can (re)install them manually with:"
+  echo "https://github.com/SumoLogic/terraform-sumologic-sumo-logic-monitor/tree/main/monitor_packages/kubernetes"
+fi
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/conf/setup/monitors.sh:59]: Please add newline at EOF

Comment (sumo-drosiek) [reply deploy/helm/sumologic/conf/setup/monitors.sh:59]: For other files as well :sweat_smile:

File: deploy/helm/sumologic/conf/setup/setup.sh [hunk 4]
```diff
@@ -142,6 +142,15 @@ TF_LOG_PROVIDER=DEBUG terraform apply \
     -var="create_fields=${CREATE_FIELDS}" \
     || { echo "Error during applying Terraform changes"; exit 1; }
 
+# Setup Sumo Logic monitors if enabled
+{{- if .Values.sumologic.setup.monitors.setupEnabled }}
+bash /etc/terraform/monitors.sh
+{{- else }}
+echo "Installation of the Sumo Logic monitors is disabled."
+echo "You can install them manually later with:"
+echo "https://github.com/SumoLogic/terraform-sumologic-sumo-logic-monitor/tree/main/monitor_packages/kubernetes"
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/conf/setup/setup.sh:151]: nobody will see this message, as setup pod is being removed just after it finish it jobs (except failures). please add this info to https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/50b1a462b19790110ecafdc33c75cd848c2fdca8/deploy/helm/sumologic/templates/NOTES.txt

File: deploy/helm/sumologic/values.yaml [hunk 5]
```diff
@@ -122,6 +122,16 @@ sumologic:
     ## uncomment for the debug mode (disables the automatic run of the setup.sh script)
     # debug: true
 
+    monitors:
+      ## If enabled, a pre-install hook will create k8s monitors in Sumo Logic
+      setupEnabled: true
+
+      ## If true, the installed monitors will be disabled by default
+      disabled: false
```
Comment (swiatekm-sumo) [inline deploy/helm/sumologic/values.yaml:130]: This set of configuration variables is a bit misleading in the context of the rest of the Chart. It's customary to have an `enabled:` key which can be used to control the whole feature. To stay consistent with that, I'd do the following:

`enabled: true` - controls the whole feature, creates monitors in setup and enables them
`monitorStatus: enabled` - sets the monitor status, can be used to disable them

Comment (starzu-sumo) [reply deploy/helm/sumologic/values.yaml:130]: OK, that makes sense. I'll update that.

File: tests/helm/terraform/static/monitors_with_email_notifications.input.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+sumologic:
+  setup:
+    monitors:
+      notificationEmails: ["test@test.lh", "email@locahost.lh"]
```
Comment (sumo-drosiek) [inline tests/helm/terraform/static/monitors_with_email_notifications.input.yaml:4]:
```suggestion
      notificationEmails: ["test@test.lh", "email@locahost.lh"]

```

File: tests/helm/terraform/static/disabled_monitors.input.yaml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+sumologic:
+  setup:
+    monitors:
+      enabled: false
+      monitorStatus: disabled
```
Comment (sumo-drosiek) [inline tests/helm/terraform/static/disabled_monitors.input.yaml:5]:
```suggestion
      monitorStatus: disabled

```

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**